packages/backend/src/exploreCompiler.ts

```diff
@@ -115,35 +115,44 @@ export const compileMetricSql = (
         ? compileMetricReference
         : compileDimensionReference;
     // Metric might have references to other dimensions
-    const renderedSql = metric.sql.replace(lightdashVariablePattern, (_, p1) =>
+    let renderedSql = metric.sql.replace(lightdashVariablePattern, (_, p1) =>
```

Comment:
I was not too sure if it is cleaner to do it this way, or wrap this function inside a compileMetricRoundSql function or something.

Let me know what's your preference